Tasks/Maven/maventask.ts

```diff
@@ -303,70 +333,108 @@ function cleanDirectory(targetDirectory:string):Q.Promise<void> {
     return <Q.Promise<void>>defer.promise;
 }
 
-// Take a result object and promise to upload the XML and HTML reports as artifacts. Throws if fail.
-function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
-    var toolStagingDir:string = path.join(stagingDir, analysisResult.toolName.toLowerCase());
-    tl.mkdirP(toolStagingDir);
+// Returns the full path of the staging directory for a given tool.
+function getStagingDirectory(toolName:string):string {
+    return path.join(stagingDir, toolName.toLowerCase());
+}
+
+// Copy output files to a staging directory and upload them as build artifacts.
+// Each module uploads its own build artifact.
+function uploadBuildArtifacts(toolName:string, moduleAnalysis:ma.ModuleAnalysis):void {
+    var localStagingDir:string = path.join(getStagingDirectory(toolName), moduleAnalysis.moduleName);
```

Comment:
Document the decision around grouping files for upload to provide a better user experience in the artefact explorer.